internal/subscription/core.go

```diff
@@ -275,3 +276,75 @@ func (c *Core) Get(ctx context.Context, key string) (*Model, error) {
 	}
 	return model, nil
 }
+
+// Migrate takes care of migrating the old subscription model to new
+func (c *Core) Migrate(ctx context.Context, names []string) error {
+
+	subscriptionsToUpdate := make([]*Model, 0)
+	if names == nil || len(names) == 0 {
+		models, err := c.List(ctx, Prefix)
```

Comment:
So if there is an empty list, all subscriptions are picked up. Currently we are assuming that there are very few subscriptions. But in case if the list is big, the API might take significant amount of time? Does this not pose any problem ?

Reply:
I had called it out during the Tech Spec discussion. Do refer `Assumptions` section for `Solution#1` under 'Migration of old subscriptions' in the Spec. This current implementation is because we know for sure the number of subscriptions is less in production.

Reply:
Have added a more detailed function description now. Do check and let me know if this suffices for the time being.